tests/integration/pilot/sidecartls/main_test.go

```diff
@@ -0,0 +1,356 @@
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+const (
+	SidecarConfig = `
+apiVersion: networking.istio.io/v1alpha3
+kind: Sidecar
+metadata:
+  name: sidecar
+  namespace:  {{.AppNamespace}}
+spec:
+  ingress:
+    - port:
+        number: 9080
+        protocol: HTTPS
+        name: custom-http
+      inboundTls:
+        mode: SIMPLE
+        serverCertificate: /etc/certs/servercert.pem
+        privateKey: /etc/certs/privatekey.pem
+      defaultEndpoint: unix:///var/run/someuds.sock
+  egress:
+    - hosts:
+      - "*/*"
+`
+	AppConfig = `
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: app
+  namespace: {{.AppNamespace}}
+spec:
+  hosts:
+  - httpbin.com
+  ports:
+  - number: 80
+    name: http
+    protocol: HTTP
+  resolution: STATIC
+  endpoints:
+  - address: 1.1.1.1
+    labels:
+      app: httpbin
+`
+
+	IncludedConfig = `
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: dependency
+  namespace: {{.AppNamespace}}
+spec:
+  hosts:
+  - app.com
+  ports:
+  - number: 80
+    name: http
+    protocol: HTTP
+  resolution: STATIC
+  endpoints:
+  - address: 2.2.2.2
+`
+	MeshConfig = `
+disablePolicyChecks: false
+mixerCheckServer: istio-policy.istio-system.svc.cluster.local:15004
+mixerReportServer: istio-telemetry.istio-system.svc.cluster.local:15004
+`
+	StricteMtls = `
+apiVersion: "security.istio.io/v1beta1"
+kind: "PeerAuthentication"
+metadata:
+  name: "default"
+  namespace: {{.AppNamespace}}
+  annotations:
+    test-suite: "beta-mtls-on"
+spec:
+  mtls:
+    mode: STRICT
+`
+	DisableMTLS = `
+apiVersion: "security.istio.io/v1beta1"
+kind: "PeerAuthentication"
+metadata:
+  name: "default"
+  namespace: {{.AppNamespace}}
+  annotations:
+    test-suite: "beta-mtls-on"
+spec:
+  mtls:
+    mode: DISABLE
+`
+)
+
+type Config struct {
+	AppNamespace string
+}
+
+func setupTest(t *testing.T, ctx resource.Context, modifyConfig func(c Config) Config) (pilot.Instance, *model.Proxy, func()) {
+	meshConfig := mesh.DefaultMeshConfig()
+	meshConfig.MixerCheckServer = "istio-policy.istio-system.svc.cluster.local:15004"
+	meshConfig.MixerReportServer = "istio-telemetry.istio-system.svc.cluster.local:15004"
+
+	g := galley.NewOrFail(t, ctx, galley.Config{MeshConfig: MeshConfig})
+	p := pilot.NewOrFail(t, ctx, pilot.Config{Galley: g, MeshConfig: &meshConfig})
+
+	appNamespace := namespace.NewOrFail(t, ctx, namespace.Config{
+		Prefix: "app",
+		Inject: true,
+	})
+
+	config := modifyConfig(Config{
+		AppNamespace: appNamespace.Name(),
+	})
+
+	// Apply all configs
+	createConfig(t, g, config, SidecarConfig, appNamespace)
+	createConfig(t, g, config, AppConfig, appNamespace)
+	createConfig(t, g, config, IncludedConfig, appNamespace)
+	createConfig(t, g, config, StricteMtls, appNamespace)
+
+	time.Sleep(time.Second * 2)
```

Comment:
We should not sleep in tests ever

Reply:
It is copied, will look if this can be removed